### PR TITLE
[SRVKE-1104] Add KafkaSink post-upgrade tests

### PR DESCRIPTION
**Pull request by pierDipi**
We include only post-upgrade tests for now, since this is the first
time we're shipping this new component.

### CODE REVIEW

**Comment by pierDipi** [comment]
```
{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"container \"verify-messagesnw2xw\" in pod \"verify-messagesnw2xw--1-w7j76\" is waiting to start: trying and failing to pull image","reason":"BadRequest","code":400}
```

File: test/upgrade/upgrade_test.go [hunk 1]
```diff
@@ -129,7 +133,9 @@ func postUpgradeTests(ctx *test.Context) []pkgupgrade.Operation {
 	tests = append(tests, eventingupgrade.PostUpgradeTests()...)
 	tests = append(tests, kafkaupgrade.ChannelPostUpgradeTest(),
 		kafkaupgrade.SourcePostUpgradeTest(),
-		kafkabrokerupgrade.BrokerPostUpgradeTest())
+		kafkabrokerupgrade.BrokerPostUpgradeTest(),
+		kafkabrokerupgrade.SinkPostDowngradeTest(),
```
Comment (mgencur) [inline test/upgrade/upgrade_test.go:137]: Why "PostDowngrade" and not "PostUpgrade" ? I can see they're same but then it's confusing in logs. We're only doing upgrades, not downgrades in Serverless operator.

Comment (pierDipi) [reply test/upgrade/upgrade_test.go:137]: Yes, `WIP`

**Comment by pierDipi** [comment]
I think the test is using the wrong image

**Comment by pierDipi** [comment]
/test unit-test

@matzew this test is based on the order of the flag value.
```
--- FAIL: TestDisabledControllers (0.00s)
    --- FAIL: TestDisabledControllers/no_broker_and_no_sink (0.00s)
        knativekafka_controller_test.go:595: Resource wasn't what we expected, diff:   &unstructured.Unstructured{
              	Object: map[string]interface{}{
              		"apiVersion": string("apps/v1"),
              		"kind":       string("Deployment"),
              		"metadata":   map[string]interface{}{"creationTimestamp": nil, "name": string("kafka-controller")},
              		"spec": map[string]interface{}{
              			"selector": nil,
              			"strategy": map[string]interface{}{},
              			"template": map[string]interface{}{
              				"metadata": map[string]interface{}{"creationTimestamp": nil},
              				"spec": map[string]interface{}{
              					"containers": []interface{}{
              						map[string]interface{}{
            - 							"args": []interface{}{
            - 								string("--disable-controllers=sink-controller,broker-controller,trigger-controller"),
            - 							},
            + 							"args": []interface{}{
            + 								string("--disable-controllers=broker-controller,trigger-controller,sink-controller"),
            + 							},
              							"name":      string("controller"),
              							"resources": map[string]interface{}{},
              						},
              					},
              				},
              			},
              		},
              		"status": map[string]interface{}{},
              	},
              }
FAIL
FAIL	github.com/openshift-knative/serverless-operator/knative-operator/pkg/controller/knativekafka	0.169s
```

**Comment by pierDipi** [comment]
Mirroring merged https://github.com/openshift/release/pull/25185

**Comment by pierDipi** [comment]
Image is there https://quay.io/repository/openshift-knative/kafka-consumer?tab=tags, let's see

/test 4.9-upgrade-tests-aws-ocp-49

**Comment by pierDipi** [comment]
This works now!

/assign @mgencur

**Comment by mgencur** [comment]
/lgtm

Great!

**Pull request merged**